app/src/lib/components/Select.svelte

```diff
@@ -27,8 +38,28 @@
 	let listOpen = false;
 	let element: HTMLElement;
 	let options: HTMLDivElement;
+	let highlightIndex: number | undefined = undefined;
+	let highlightedItem: Selectable | undefined = undefined;
+	let filterText: string | undefined = undefined;
+	let filteredItems: Selectable[] = items;
+
+	function filterItems(items: Selectable[], filterText: string | undefined) {
+		if (!filterText) {
+			return items;
+		}
+
+		return items.filter((it) => {
+			const property = it[labelId];
+			if (!isStr(property)) return false;
+			return property.includes(filterText);
+		});
+	}
 
-	function handleItemClick(item: any) {
+	$: filteredItems = filterText === undefined ? items : filterItems(items, filterText);
```

Comment:
filterItems handles the case for when the filterText is undefined so this can be reduced to the following:
```suggestion
$: filteredItems = filterItems(items, filterText);
```

Reply:
Oh right, nice catch!